internal/handlers.go

```diff
@@ -0,0 +1,81 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+func Index(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	fmt.Fprint(w, "Welcome to GoPay!")
+}
+
+func GetAllAccounts(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	accs := []*Account{}
+
+	for _, account := range Accounts {
+		accs = append(accs, account)
+	}
+
+	response := &JsonResponse{Data: &accs}
+	w.Header().Set("Content-Type", "application/json; charset=UTF8")
+	w.WriteHeader(http.StatusOK)
```

Comment:
Why are we writing the header for `OK` before we encode the response?

We should have this in this order:
```go
if err := json.NewEncoder(w).Encode(response); err != nil {
    w.WriteHeader(http.StatusInternalServerError)
    json.NewEncoder(w).Encode(&JsonErrorResponse{Error: &ApiError{Status: http.StatusNotFound, Title: "Internal Server Error"}})
    return
}

response := &JsonResponse{Data: &accs}
w.Header().Set("Content-Type", "application/json; charset=UTF8")
w.WriteHeader(http.StatusOK)
```